(Actions) Conversational AI (#31390)
DataDog/documentation: content/en/actions/app_builder/build.md

1. Each blueprint template comes loaded with demo data. To customize the app, edit the **Connection** for each query.
1. To save the app, click **Save as New App**.
1. To preview the app, click **Preview**. Click **Edit** from the preview screen to return to the configuration view.
1. When you're ready to publish your app, click **Publish**. Publishing an app makes it available to your dashboards.

### Build an app with AI

> With Bits AI, you can generate a complete app from a single prompt. Describe the app you want to create, and the agent will automatically generate the UI, actions, and logic mapped to your existing data and permissions. You can then iterate by chatting---refining components, flows, or styling---all without writing any code.

Reviewer: ```suggestion
With Bits AI, you can generate a complete app from a single prompt. Describe the app you want to create, and the agent automatically generates the UI, actions, and logic mapped to your existing data and permissions. You can then iterate by chatting---refining components, flows, or styling---without writing any code.
```

Reviewer: I'm guessing `agent` here does not refer to the Datadog Agent. Would it be possible to disambiguate this, as references to the Datadog Agent are widespread in the docs?

1. After you finish modifying the app, Click **Run** to test it.
1. When you're ready to publish your app, click **Publish**. Publishing an app makes it available to your dashboards.

### Build an app with AI

You can use Bits AI to create an app. To get started:
1. From [App Builder][1], click **New App**.
1. Click **Start with AI**.
1. Click a suggested prompt, or enter a prompt that describes the app you want to create. Enter as much detail as possible to improve the results. Here are some example queries:
   - `Display a list of lambda functions in a table. Allow the user to filter by function name and specify the limit.`
   - `Provide a list of PagerDuty services and their status in a table. Include a search input above the table.`
   - `Display GitHub pull requests in a table.`

> 1. After Bits AI responds to your prompt, you can click **Edit with AI** or manually [customize your app][15].

Reviewer 2: Before that, can you add the following steps:
1. Explain that the agent will will auto-generate the UI, actions, and logic, mapped to the customer's existing data and permissions.
2. explain about the step where the agent ask the user User to set up a connection to build the app with real data or skip and build the app layout faster without data
3. Mention that in some cases, the agent will ask clarification questions.

> 1. By default, App Builder saves automatically. You can disable auto-save by clicking the gear <i class="icon-cog-2"></i> icon. To save the app manually, click **Save**.

Reviewer 2: Why is it relevant here? Also, I think the auto-save kicks in only after the first time the user saves the app

1. After you finish modifying the app, Click **Run** to test it.
1. When you're ready to publish your app, click **Publish**. Publishing an app makes it available to your dashboards.

> ### Build an app with AI

Reviewer 2: Can we make this the main option? i.e change the order between "build an app with AI" to "Build an app from a blueprint"

You can use Bits AI to create an app. To get started:
1. From [App Builder][1], click **New App**.
1. Click **Start with AI**.
1. Click a suggested prompt, or enter a prompt that describes the app you want to create. Enter as much detail as possible to improve the results. Here are some example queries:
   - `Display a list of lambda functions in a table. Allow the user to filter by function name and specify the limit.`

>    - `Provide a list of PagerDuty services and their status in a table. Include a search input above the table.`

Reviewer: ```suggestion
   - `Provide a list of PagerDuty services and their statuses in a table. Include a search input above the table.`
```

   - `Display GitHub pull requests in a table.`
1. Click the up arrow (**↑**) to send your prompt. Bits AI will automatically generate the UI, actions, and logic for your app, mapped to your existing data and permissions.

> 1. While it's responding, Bits AI will ask you to set up a connection to build the app with real data. You can complete this step, or skip it to build the app layout faster without data.

Reviewer: ```suggestion
1. While it's responding, Bits AI asks you to set up a connection to build the app with real data. You can complete this step, or skip it to build the app layout faster without data.
```

1. After you finish modifying the app, Click **Run** to test it.
1. When you're ready to publish your app, click **Publish**. Publishing an app makes it available to your dashboards.

### Build an app with AI

With Bits AI, you can generate a complete app from a single prompt. Describe the app you want to create, and the agent will automatically generate the UI, actions, and logic mapped to your existing data and permissions. You can then iterate by chatting---refining components, flows, or styling---all without writing any code.

To build an app with Bits AI:
1. From [App Builder][1], click **New App**.
1. Click **Start with AI**.
1. Click a suggested prompt, or enter a prompt that describes the app you want to create. Enter as much detail as possible to improve the results. Here are some example queries:
   - `Display a list of lambda functions in a table. Allow the user to filter by function name and specify the limit.`

>    - `Provide a list of PagerDuty services and their status in a table. Include a search input above the table.`

Reviewer 3: @JohanDelao thank you for catching that! i've cut it from the doc. do you have a suggestion for a replacement example query? no worries if not

   - `Display GitHub pull requests in a table.`
1. Click the up arrow (**↑**) to send your prompt. Bits AI will automatically generate the UI, actions, and logic for your app, mapped to your existing data and permissions.
1. While it's responding, Bits AI will ask you to set up a connection to build the app with real data. You can complete this step, or skip it to build the app layout faster without data.

> 1. Bits AI sometimes asks clarification questions while it responds to your prompt. If this happens, enter a response, then click the up arrow (**↑**). 

Reviewer 4: it could be good to add a bullet here about mocking with AI

```
After Bits AI adds an action to the app, you will be prompted to either Mock with AI Data, or skip configuration. Configuring your action during agent execution helps with connecting the data types to other components.
```

Reviewer 3: @setnilson great suggestion! added. thank you!

1. After Bits AI responds to your prompt, you can click **Edit with AI** or manually [customize your app][15].
1. To save the app, click **Save**.
1. To preview the app, click **View**. Click **Edit** to return to the configuration view.
1. After you finish modifying the app, click **Run** to test it.
1. When you're ready to publish your app, click **Publish**. Publishing an app makes it available to your dashboards.

To iterate on an existing app:
1. From an existing app, click **UI Agent Chat** (**<i class="icon-bits-ai"></i>**).
1. Enter a detailed prompt for the behavior you'd like to add to your app. Include the integrations and actions you'd like to use.

> 1. Click the up arrow (**↑**) to add the functionality to your app.

Reviewer 3: ```suggestion
1. Press **Enter** to add the functionality to your app.
```
